packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -211,13 +236,53 @@ export function useConnections({
   );
   const { activeConnectionId } = state;
 
+  const getActiveAndInactiveCount = useCallback(() => {
+    const total = favoriteConnections.length + recentConnections.length;
+    const active =
+      connectionsManager.getConnectionIdsByStatus(ConnectionStatus.Connected)
+        .length || 0;
+    return { active, inactive: total - active };
+  }, [favoriteConnections, recentConnections, connectionsManager]);
+
+  const findConnectionInfo = useCallback(
+    (connectionId: ConnectionInfo['id']) => {
+      return [...favoriteConnections, ...recentConnections].find(
+        ({ id }) => id === connectionId
+      );
+    },
+    [favoriteConnections, recentConnections]
+  );
+
   const saveConnectionInfo = useCallback(
-    async (
-      connectionInfo: RecursivePartial<ConnectionInfo> &
-        Pick<ConnectionInfo, 'id'>
-    ) => {
+    async ({
```

Comment:
can you elaborate on this change? what made us change how connections are saved here?

Reply:
we don't change how they are saved, we're adding tracking for connections that are new. this method is used for create & update both. in one case of update, it is called with partial connectionInfo, which doesn't fit the tracking.

separating create & update would be one option, it just doesn't fit with the current flow. I also didn't want to simply assume that the info is complete when the connection is new, so I opted to make the `partialConnectionInfo` an explicit overload.

Reply:
Got it, this is creating new concepts though, `partialConnectionInfo` and `fullConnectionInfo` isn't that obvious in which way would be full and partial. If we want to keep this style, it may be better to add some jsdoc comment that clarifies how this function should be used, when you want to call what.

> I also didn't want to simply assume that the info is complete when the connection is new

This is a good example, i feel we are giving for granted that everyone understands what "complete" means here.

I also want to add that I do feel weird that we need to do an invasive change like this for tracking. I wonder what other options we considered

Reply:
the change doesn't seem that invasive to me - it's a change in parameters of a function internal to `useConnections` hook

Reply:
I think I listed all the alternatives above. From those, separating create & update would definitely be more invasive, which leaves the alternative to "assume" that a new connection info is complete (with `as`). This is mildly less invasive (doesn't change the parameters of the internal function), but less secure.
perhaps you could check the change in the function again? now that we've removed the active/inactive count, the change is leaner.